### PR TITLE
Make profile screen

### PR DESCRIPTION
**Pull request by OgnyanY**
Simulator Screenshot - iPhone 14 Pro - 2024-02-19 at 16 03 13

### CODE REVIEW

File: DigitalLibrary/DigitalLibrary/MainScreens/ProfileScreen/ProfileViewModel.swift [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+            updatingUser.email = self.email
+        case .firstName:
+            updatingUser.firstName = self.firstName
+        case .lastName:
+            updatingUser.lastName = self.lastName
+        case .phone:
+            updatingUser.phone = self.phone
+        }
+
+        print("User data updated successfully")
```
Comment (vasilkadt) [inline DigitalLibrary/DigitalLibrary/MainScreens/ProfileScreen/ProfileViewModel.swift]:
```suggestion

```

File: DigitalLibrary/DigitalLibrary/MainScreens/ProfileScreen/ProfileViewModel.swift [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+                    self?.lastName = fetchedUser.lastName
+                    self?.phone = fetchedUser.phone
+                }
+            }
+        }
+    }
+
+    func updateUserData(field: ProfileField) {
+        guard var updatingUser = user else {
+            print("Failed to update user data")
```
Comment (vasilkadt) [inline DigitalLibrary/DigitalLibrary/MainScreens/ProfileScreen/ProfileViewModel.swift]: We do not print, we show alerts. Check the other classes

File: DigitalLibrary/DigitalLibrary/MainScreens/ProfileScreen/ProfileViewModel.swift [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+        self.authenticationProvider = authenticationProvider
+        self.usersProvider = usersProvider
+    }
+
+    func fetchCurrentUser() {
+        Task {
+            do {
+                guard let fetchedUser = await usersProvider.getCurrentUser() else {
+                    // Handle the case where no user is fetched
+                    print("No user data was fetched.")
```
Comment (vasilkadt) [inline DigitalLibrary/DigitalLibrary/MainScreens/ProfileScreen/ProfileViewModel.swift]: Same here, we want to show alert with the error

File: DigitalLibrary/DigitalLibrary/MainScreens/ProfileScreen/ProfileViewModel.swift [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+//  ProfileViewModel.swift
+//  DigitalLibrary
+//
+//  Created by Ognyan Yonchev on 16.02.24.
+//
+
+import SwiftUI
+
+final class ProfileViewModel: ObservableObject {
+    //@ObservedObject private var appRootManager: AppRootManager
```
Comment (vasilkadt) [inline DigitalLibrary/DigitalLibrary/MainScreens/ProfileScreen/ProfileViewModel.swift:11]: Is this needed?

File: DigitalLibrary/DigitalLibrary/MainScreens/ProfileScreen/ProfileView.swift [hunk 5]
```diff
@@ -8,7 +8,64 @@
...
+                    .clipped()
+                    .cornerRadius(50)
+
+                Text("Profile")
+                    .font(.largeTitle)
+                    .fontWeight(.bold)
+            }
+            .frame(maxWidth: .infinity)
+
+            // Fields
```
Comment (vasilkadt) [inline DigitalLibrary/DigitalLibrary/MainScreens/ProfileScreen/ProfileView.swift]: No need for this comments

File: DigitalLibrary/DigitalLibrary/MainScreens/ProfileScreen/ProfileView.swift [hunk 6]
```diff
@@ -8,7 +8,64 @@
...
 
 struct ProfileView: View {
+    @ObservedObject var viewModel: ProfileViewModel
+
     var body: some View {
-        Text("Profile")
+        VStack {
+            Spacer(minLength: 0)
+
+            // Profile title and image centered
```
Comment (vasilkadt) [inline DigitalLibrary/DigitalLibrary/MainScreens/ProfileScreen/ProfileView.swift]:
```suggestion

```

File: DigitalLibrary/DigitalLibrary/MainScreens/ProfileScreen/ProfileInfoFieldView.swift [hunk 7]
```diff
@@ -0,0 +1,63 @@
...
+struct ProfileInfoField: View {
+    let label: String
+    @Binding var text: String
+    @Binding var editingField: ProfileField?
+    let fieldType: ProfileField
+    let onCommit: () -> Void
+
+    var isEditing: Bool {
+        editingField == fieldType
+    }
```
Comment (vasilkadt) [inline DigitalLibrary/DigitalLibrary/MainScreens/ProfileScreen/ProfileInfoFieldView.swift]: This should be in viewModel

File: DigitalLibrary/DigitalLibrary/MainScreens/ProfileScreen/ProfileViewModel.swift (hunk 2 above)
Comment (OgnyanY) [reply DigitalLibrary/DigitalLibrary/MainScreens/ProfileScreen/ProfileViewModel.swift]: ok i will change it

File: DigitalLibrary/DigitalLibrary/MainScreens/ProfileScreen/ProfileViewModel.swift (hunk 4 above)
Comment (OgnyanY) [reply DigitalLibrary/DigitalLibrary/MainScreens/ProfileScreen/ProfileViewModel.swift:11]: I will use it for the sign out button

File: DigitalLibrary/DigitalLibrary/MainScreens/ProfileScreen/ProfileViewModel.swift (hunk 3 above)
Comment (OgnyanY) [reply DigitalLibrary/DigitalLibrary/MainScreens/ProfileScreen/ProfileViewModel.swift]: ok

File: DigitalLibrary/DigitalLibrary/MainScreens/ProfileScreen/ProfileView.swift (hunk 5 above)
Comment (OgnyanY) [reply DigitalLibrary/DigitalLibrary/MainScreens/ProfileScreen/ProfileView.swift]: Ok

File: DigitalLibrary/DigitalLibrary/MainScreens/ProfileScreen/ProfileView.swift (hunk 6 above)
Comment (OgnyanY) [reply DigitalLibrary/DigitalLibrary/MainScreens/ProfileScreen/ProfileView.swift]: I will remove it

File: DigitalLibrary/DigitalLibrary/MainScreens/ProfileScreen/ProfileInfoFieldView.swift (hunk 7 above)
Comment (OgnyanY) [reply DigitalLibrary/DigitalLibrary/MainScreens/ProfileScreen/ProfileInfoFieldView.swift]: ok

File: DigitalLibrary/DigitalLibrary/MainScreens/ProfileScreen/ProfileViewModel.swift (hunk 1 above)
Comment (OgnyanY) [reply DigitalLibrary/DigitalLibrary/MainScreens/ProfileScreen/ProfileViewModel.swift]: I will remove it

File: DigitalLibrary/DigitalLibrary/MainScreens/ProfileScreen/ProfileView.swift [hunk 8]
```diff
@@ -8,7 +8,57 @@
...
+            ProfileInfoField(viewModel: .init(label: "Phone",
+                                              text: $viewModel.phone,
+                                              fieldType: .phone,
+                                              editingField: $viewModel.editingField,
+                                              onCommit: { viewModel.updateUserData(field: .phone) }))
+        }
+        .padding()
+        .onAppear {
+            viewModel.fetchCurrentUser()
+            print("Email\(viewModel.email)Name\(viewModel.firstName)\(viewModel.lastName)Phone\(viewModel.phone)")
```
Comment (vasilkadt) [inline DigitalLibrary/DigitalLibrary/MainScreens/ProfileScreen/ProfileView.swift]:
```suggestion

```

File: DigitalLibrary/DigitalLibrary/MainScreens/ProfileScreen/ProfileViewModel.swift (hunk 4 above)
Comment (vasilkadt) [reply DigitalLibrary/DigitalLibrary/MainScreens/ProfileScreen/ProfileViewModel.swift:11]: ок

File: DigitalLibrary/DigitalLibrary/MainScreens/ProfileScreen/ProfileViewModel.swift [hunk 9]
```diff
@@ -0,0 +1,73 @@
...
+                self.email = fetchedUser.email
+                self.firstName = fetchedUser.firstName
+                self.lastName = fetchedUser.lastName
+                self.phone = fetchedUser.phone
+            }
+        }
+    }
+
+    func updateUserData(field: ProfileField) {
+        guard var updatingUser = user else {
```
Comment (vasilkadt) [inline DigitalLibrary/DigitalLibrary/MainScreens/ProfileScreen/ProfileViewModel.swift]: I think updatedUser is better

File: DigitalLibrary/DigitalLibrary/MainScreens/ProfileScreen/ProfileView.swift (hunk 8 above)
Comment (OgnyanY) [reply DigitalLibrary/DigitalLibrary/MainScreens/ProfileScreen/ProfileView.swift]: done

File: DigitalLibrary/DigitalLibrary/MainScreens/ProfileScreen/ProfileViewModel.swift (hunk 9 above)
Comment (OgnyanY) [reply DigitalLibrary/DigitalLibrary/MainScreens/ProfileScreen/ProfileViewModel.swift]: done

**Review by vasilkadt — APPROVED** [review]

**Pull request merged**